muselog/datadog.py

```diff
@@ -77,10 +78,35 @@ def default(self, obj):
 
 class DatadogJSONFormatter(json_log_formatter.JSONFormatter):
 
+    trace_enabled = False
+
+    def __init__(self, trace_enabled=False):
+        self.trace_enabled = trace_enabled
+
+    def inject_trace_values(self, record):
+        """
+        Injects logs with a 'trace_id' and 'span_id'. If a trace is active this helps DD
+        to correlate logs sent to that specific trace in APM.
+        """
+        if not self.trace_enabled:
+            return record
+
+        # Create a new record so we don't modify the original
+        new_record = record.copy()
+
+        # get correlation ids from current tracer context
+        trace_id, span_id = helpers.get_correlation_ids()
+
+        new_record['dd.trace_id'] = trace_id or 0
```

Comment:
is `0` the suggested id value when there's no id to provide, or just something you decided to use?

Reply:
It's the suggested value, I took that snippet from here https://docs.datadoghq.com/tracing/advanced_usage/?tab=python#correlate-traces-and-logs